### ISSUE TITLE
Can't run on GPU

### ISSUE THREAD

**Issue by gaopeng990618** [georgepar/mmlatch#4]
Hi thank you so much for sharing your work. I am trying to recreate the results. I am using the CMU-MOSEi dataset you provided . The code has an error on GPU, but no problem on CPU. So I find out in rnn.py has some issue so I fix out to make sure **x and lengths** are in the same device:
```
class PackSequence(nn.Module):
    def __init__(self, batch_first=True, device="cpu",):
        super(PackSequence, self).__init__()
        self.batch_first = batch_first

        self.device = device

    def forward(self, x, lengths):
        x = pack_padded_sequence(
            x, lengths, batch_first=self.batch_first, enforce_sorted=False
        )
        lengths = lengths.to(self.device)
        lengths = lengths[x.sorted_indices]

        return x, lengths
```
But I get NaN  out from Train Loss . Could you please suggest why this happened.
```
return loss_value:4.928342342376709
Epoch [1/100]: [1/511]   0%|                         , Train Loss=4.93 [00:00<?]
return loss_value:nan
Epoch [1/100]: [2/511]   0%|                      , Train Loss=nan [00:00<00:54]
return loss_value:nan
Epoch [1/100]: [2/511]   0%|                      , Train Loss=nan [00:00<00:54]
return loss_value:nan
Epoch [1/100]: [4/511]   1%|▏                     , Train Loss=nan [00:00<00:42]
return loss_value:nan
Epoch [1/100]: [4/511]   1%|▏                     , Train Loss=nan [00:00<00:42]
return loss_value:nan
Epoch [1/100]: [6/511]   1%|▎                     , Train Loss=nan [00:00<00:38]
return loss_value:nan
Epoch [1/100]: [6/511]   1%|▎                     , Train Loss=nan [00:00<00:38]
```

**Comment by Jockzhk** [comment]
Hello, have you solved it

**Comment by Galaxy-219** [comment]
`if __name__ == "__main__":
    cfg = load_config("../../tests/test.yaml")
    import pprint
    pprint.pprint(cfg)`The code cannot run, it's missing the 'test' file. Do you have the same issue?

**Comment by Galaxy-219** [comment]
> 嗨，非常感谢你分享你的作品。我正试图重现结果。我正在使用您提供的CMU-MOSEi数据集。代码在GPU上有一个错误，但在CPU上没有问题。所以我在rnn.py中发现一些问题，所以我修复了，以确保**和长度**在同一设备中：>> `````>类PackSequence（nn.Module）：> def __init__(self, batch_first=True, device="cpu",):> super(PackSequence, self).__init__()> self.batch_first = batch_first>> self.device = 设备>> def forward（self，x，lengths）：> x = pack_padded_sequence(> x，长度，batch_first=self.batch_first，force_sorted=False> ）> lengths = lengths.to（self.device）> 长度 = [x.sorted_indices]>>返回x，长度，长度> `NaN``我``我可以从我從>>火车损失中损失中出来你能建议一下为什么会发生这种情况吗？>> ```>返回损失_价值：4.928342342376709> Epoch [1/100]：[1/511] 0%|，火车损失=4.93 [00:00<?]>返回损失_值：nan> Epoch [1/100]: [2/511] 0%| , Train Loss=nan [00:00<00:54]>返回损失_值：nan> Epoch [1/100]: [2/511] 0%| , Train Loss=nan [00:00<00:54]>返回损失_值：nan> Epoch [1/100]: [4/511] 1%|▏ , Train Loss=nan [00:00<00:42]>返回损失_值：nan> Epoch [1/100]: [4/511] 1%|▏ , Train Loss=nan [00:00<00:42]>返回损失_值：nan> 大纪元 [1/100]: [6/511] 1%|▎ , Train Loss=nan [00:00<00:38]>返回损失_值：nan> 大元 [6/511] 1%|纪▎ , Train Loss> 嗨，非常感谢你分享你的作品。我正试图重现结果。我正在使用您提供的CMU-MOSEi数据集。代码在GPU上有一个错误，但在CPU上没有问题。所以我在rnn.py中发现一些问题，所以我修复了，以确保**x和长度**在同一设备中：>> ```>类PackSequence（nn.Module）：> def __init__(self, batch_first=True, device="cpu",):> super(PackSequence, self).__init__()> self.batch_first = batch_first>> self.device = 设备>> def forward（self，x，lengths）：> x = pack_padded_sequence(> x，长度，batch_first=self.batch_first，force_sorted=False> ）> lengths = lengths.to（self.device）> 长度 = 长度[x.sorted_indices]>>返回x，长度> ```>> 但我让NaN从火车损失中出来。你能建议一下为什么会发生这种情况吗？>> ```>返回损失_价值：4.928342342376709> Epoch [1/100]：[1/511] 0%|，火车损失=4.93 [00:00<?]>返回损失_值：nan> Epoch [1/100]: [2/511] 0%| , Train Loss=nan [00:00<00:54]>返回损失_值：nan> Epoch [1/100]: [2/511] 0%| , Train Loss=nan [00:00<00:54]>返回损失_值：nan> Epoch [1/100]: [4/511] 1%|▏ , Train Loss=nan [00:00<00:42]>返回损失_值：nan> Epoch [1/100]: [4/511] 1%|▏ , Train Loss=nan [00:00<00:42]>返回损失_值：nan> 大纪元 [1/100]: [6/511] 1%|▎ , Train Loss=nan [00:00<00:38]>返回损失_值：nan> 大纪元 [1/100]: [6/511] 1%|▎ , Train Loss=nan [00:00<00:38]> ```请问你是否遇到缺少test文件导致config.py无法运行的情况

**Comment by Jamescqw** [comment]
> 嗨，非常感谢您分享您的工作。我正在尝试重新创建结果。我正在使用您提供的 CMU-MOSEi 数据集。代码在 GPU 上有错误，但在 CPU 上没有问题。所以我发现 rnn.py 有一些问题，所以我修复以确保 **x 和长度**在同一个设备中：
> 
> ```
> class PackSequence(nn.Module):
>     def __init__(self, batch_first=True, device="cpu",):
>         super(PackSequence, self).__init__()
>         self.batch_first = batch_first
> 
>         self.device = device
> 
>     def forward(self, x, lengths):
>         x = pack_padded_sequence(
>             x, lengths, batch_first=self.batch_first, enforce_sorted=False
>         )
>         lengths = lengths.to(self.device)
>         lengths = lengths[x.sorted_indices]
> 
>         return x, lengths
> ```
> 
> 但是我从火车损失中得到了 NaN.你能说出为什么会这样吗？
> 
> ```
> return loss_value:4.928342342376709
> Epoch [1/100]: [1/511]   0%|                         , Train Loss=4.93 [00:00<?]
> return loss_value:nan
> Epoch [1/100]: [2/511]   0%|                      , Train Loss=nan [00:00<00:54]
> return loss_value:nan
> Epoch [1/100]: [2/511]   0%|                      , Train Loss=nan [00:00<00:54]
> return loss_value:nan
> Epoch [1/100]: [4/511]   1%|▏                     , Train Loss=nan [00:00<00:42]
> return loss_value:nan
> Epoch [1/100]: [4/511]   1%|▏                     , Train Loss=nan [00:00<00:42]
> return loss_value:nan
> Epoch [1/100]: [6/511]   1%|▎                     , Train Loss=nan [00:00<00:38]
> return loss_value:nan
> Epoch [1/100]: [6/511]   1%|▎                     , Train Loss=nan [00:00<00:38]
> ```

Hello, I have also encountered the same problem as you. Have you resolved it?
